core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -1107,6 +1108,16 @@ object UniqueTriggerActivation {
                             otherCiv.getDiplomacyManagerOrMeet(civInfo).addModifier(DiplomaticModifiers.StealingTerritory, -10f)
                             civsToNotify.add(otherCiv)
                         }
+                        // check if civ has steal a tile from a citystate 
+                        if (otherCiv != null && otherCiv.isCityState) {
+                            // create this varibale diplomacyCityState for more readability
+                            val diplomacyCityState = otherCiv.getDiplomacyManager(civInfo)!!
```

Comment:
How about getDiplomacyManagerOrMeet, that way we don't run into weird problems if we manage to steal tiles from a civ we haven't met yet